cylc/flow/parsec/validate.py

```diff
@@ -1007,22 +1007,30 @@ def coerce_parameter_list(cls, value, keys):
             >>> CylcConfigValidator.coerce_parameter_list('a, b, c', None)
             ['a', 'b', 'c']
 
+            >>> CylcConfigValidator.coerce_parameter_list('084_132', None)
+            ['084_132']
+
+            >>> CylcConfigValidator.coerce_parameter_list('72, a', None)
+            ['72', 'a']
         """
         items = []
         can_only_be = None   # A flag to prevent mixing str and int range
         for item in cls.strip_and_unquote_list(keys, value):
             values = cls.parse_int_range(item)
             if values is not None:
-                if can_only_be == str:
+                if can_only_be is str:
                     raise IllegalValueError(
                         'parameter', keys, value, 'mixing int range and str')
                 can_only_be = int
                 items.extend(values)
             elif cls._REC_NAME_SUFFIX.match(item):  # noqa: SIM106
                 try:
+                    if '_' in item:
+                        raise ValueError(
+                            'Artifical value error undoing PEP-515')
```

Comment:
I don't understand this bit

Reply:
If there is an underscore in the string we don't want it treated as a number but to skip to line 1032, so we artificially raise a value error. 

Since it is in a try-except loop, the only purpose of the string is explanatory.

Reply:
Ah got it. Perhaps
```suggestion
                    if '_' in item:
                        # Disable PEP-515 int coercion; go to except block:
                        raise ValueError()
```